### DISCUSSION TITLE
A fatal esptool.py error occurred: Write timeout

### DISCUSSION THREAD

**Discussion by prithwirajbose** [esp8266/Arduino#8932] [Q&A]
I have constantly been getting this error while trying to upload a sketch on Wemos D1 Mini R1 & R2 type board. 
`A fatal esptool.py error occurred: Write timeout`

I've recently updated my Arduino IDE, so I assumed that it might be due to a bad installation. But after reinstalling everything including the Board Manager lib for ESP, Arduino IDE itself nothing worked. When I click on the Board Info, it displays an error "Native serial port, can't obtain info". I've tried reinstalling the CH340 driver. Practically nothing has worked so far.

I've also tried different type of ESP boards (e.g. NodeMCU) but for all boards the problem is same.

Following is the full trace of log getting printed:

```
. Variables and constants in RAM (global, static), used 28104 / 80192 bytes (35%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ DATA     1496     initialized variables
╠══ RODATA   920      constants       
╚══ BSS      25688    zeroed variables
. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 59667 / 65536 bytes (91%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ ICACHE   32768    reserved space for flash instruction cache
╚══ IRAM     26899    code in IRAM    
. Code in flash (default, ICACHE_FLASH_ATTR), used 232148 / 1048576 bytes (22%)
║   SEGMENT  BYTES    DESCRIPTION
╚══ IROM     232148   code in flash   
esptool.py v3.0
Serial port COM8
Connecting...

A fatal esptool.py error occurred: Write timeout
```

**Comment by anibabajanian** [comment]
Hello, have you found a solution to this problem?

**Comment by Satyamjp** [comment]
same error 

Variables and constants in RAM (global, static), used 37020 / 80192 bytes (46%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ DATA     1628     initialized variables
╠══ RODATA   3480     constants       
╚══ BSS      31912    zeroed variables
. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 60855 / 65536 bytes (92%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ ICACHE   32768    reserved space for flash instruction cache
╚══ IRAM     28087    code in IRAM    
. Code in flash (default, ICACHE_FLASH_ATTR), used 509500 / 1048576 bytes (48%)
║   SEGMENT  BYTES    DESCRIPTION
╚══ IROM     509500   code in flash   
esptool.py v3.0
Serial port COM3
Connecting......................................_____
A fatal esptool.py error occurred: Failed to connect to ESP8266: Timed out waiting for packet header
